### PR TITLE
Update expiration of current contestants in contestSetAdditionalTime

### LINKED ISSUE #282: Update

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #282

### CODE REVIEW

File: app/api/contests/set_additional_time.go [hunk 1]
```diff
@@ -57,59 +61,132 @@ import (
...
+		service.MustNotBeError(scope.UpdateColumn("additional_time",
+			gorm.Expr("SEC_TO_TIME(?)", additionalTimeInSeconds)).Error())
+	} else if additionalTimeInSeconds != 0 {
+		service.MustNotBeError(groupContestItemStore.Exec(
+			"INSERT INTO groups_contest_items (group_id, item_id, additional_time) VALUES(?, ?, SEC_TO_TIME(?))",
+			groupID, itemID, additionalTimeInSeconds).Error())
+	}
+
+	service.MustNotBeError(store.Exec("DROP TEMPORARY TABLE IF EXISTS total_additional_times").Error())
+	service.MustNotBeError(store.Exec(`
```
Comment (smadbe) [inline app/api/contests/set_additional_time.go]: This request probably needs some explanations (comments), especially as there are many kind of groups and it is very hard to follow.
For instance, indicating after L157 that, at this point, we have all groups participating to the contest and which are descendant of the group on which we change additional time (so, that will be affected)

File: app/api/contests/set_additional_time.go [hunk 2]
```diff
@@ -57,59 +61,132 @@ import (
...
+					ON groups_contest_items.group_id = groups_ancestors.ancestor_group_id AND
+						groups_contest_items.item_id = contest_participations.item_id`).
+			Group("groups_groups.child_group_id").
+			Select(`
+				groups_groups.child_group_id,
+				IFNULL(SUM(TIME_TO_SEC(groups_contest_items.additional_time)), 0) AS total_additional_time`).
+			WithWriteLock().QueryExpr()).Error())
+
+	result := store.Exec(`
+		UPDATE groups_groups
```
Comment (smadbe) [inline app/api/contests/set_additional_time.go]: If I do the UPDATE using the SELECT request directly inside it, it seems to works on my db. (so no needing the temp table) I just replaced `JOIN total_additional_times` by `JOIN (SELECT ... ) AS total_additional_times`
Could you check? 
If so, probably there are also ways to write it in a bit cleaner way.

Comment (zenovich) [reply app/api/contests/set_additional_time.go]: Do you have triggers in your DB? (Initially I implemented this with nested select, but that didn't work)

Comment (zenovich) [reply app/api/contests/set_additional_time.go]: ("Can't update table in stored function/trigger because it is already used by statement which invoked this stored function/trigger")

File: app/api/contests/set_additional_time.go (hunk 1 above)
Comment (zenovich) [reply app/api/contests/set_additional_time.go]: fixed

File: app/api/contests/set_additional_time.go (hunk 2 above)
Comment (smadbe) [reply app/api/contests/set_additional_time.go]: Yes, I have the triggers and applied the migrations, strange. 🤔 The tables are empty, but this error should not depend on the actual data.

Comment (zenovich) [reply app/api/contests/set_additional_time.go]: Just tried this again: ```Data: {"success":false,"message":"Internal Server Error","error_text":"Error 1442: Can't update table 'groups_ancestors' in stored function/trigger because it is already used by statement which invoked this stored function/trigger."}```
